rule/utils_test.go

```diff
@@ -30,3 +31,195 @@ func TestIsCallToExitFunction(t *testing.T) {
+		{"💡", false}, // Emoji
+		{"你", false}, // Chinese character
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.varName, func(t *testing.T) {
+			if got := hasUpperCaseLetter(tt.varName); got != tt.expected {
+				t.Errorf("hasCaps(%s) = %v; want %v", tt.varName, got, tt.expected)
+			}
+		})
+	}
+}
+
+func BenchmarkHasUpperCaseRE(b *testing.B) {
+	var anyUpperCaseRE = regexp.MustCompile(`[A-Z]`)
+	for i := 0; i < b.N; i++ {
+		input := "HeLlo_WoRlD"
+		_ = anyUpperCaseRE.MatchString(input)
+	}
+}
+
+func BenchmarkHasUpperCase(b *testing.B) {
+	for i := 0; i < b.N; i++ {
+		input := "HeLlo_WoRlD"
+		_ = hasUpperCaseLetter(input)
+	}
+}
+
+func BenchmarkAllCapsRE(b *testing.B) {
+	var allUpperCaseRE = regexp.MustCompile(`^_?[A-Z][A-Z\d]*(_[A-Z\d]+)*$`)
+	for i := 0; i < b.N; i++ {
+		input := "_SOME_PRIVATE_CONST_2"
+		_ = allUpperCaseRE.MatchString(input)
+	}
+}
+
+func BenchmarkAllCaps(b *testing.B) {
+	for i := 0; i < b.N; i++ {
+		input := "_SOME_PRIVATE_CONST_2"
+		_ = hasUpperCaseLetter(input)
+	}
+}
+
+func TestIsUpperConstFunction(t *testing.T) {
+	tests := []struct {
+		varName  string
+		expected bool
+	}{
+		{"SOME_CONST_2", true},
+		{"__FOO", false},
+		{"__", false},
+		{"X509B", true},
+		{"FOO", true},
+		{"1FOO", false},
+		{"_FOO123_BAR456", true},
+		{"A1_B2_C3", true},
+		{"A1_b2", false},
+		{"FOO_", false},
+		{"foo", false},
+		{"_", false},
+		{"", false},
+		{"FOOBAR", true},
+		{"FO", true},
+		{"F_O", true},
+		{"FOO123", true},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.varName, func(t *testing.T) {
+			if got := isUpperCaseConst(tt.varName); got != tt.expected {
+				t.Errorf("isUpperCaseConst(%s) = %v; want %v", tt.varName, got, tt.expected)
+			}
+		})
+	}
+}
+
+func BenchmarkUpperCaseConstRE(b *testing.B) {
+	var upperCaseConstRE = regexp.MustCompile(`^_?[A-Z][A-Z\d]*(_[A-Z\d]+)*$`)
+	for i := 0; i < b.N; i++ {
+		input := "A1_B2_C3"
+		_ = upperCaseConstRE.MatchString(input)
+	}
+}
+
+func BenchmarkIsUpperCaseConst(b *testing.B) {
+	for i := 0; i < b.N; i++ {
+		input := "A1_B2_C3"
+		_ = isUpperCaseConst(input)
+	}
+}
+
+func TestIsUpperUnderScoreFunction(t *testing.T) {
+	tests := []struct {
+		varName  string
+		expected bool
+	}{
+		{"_", false},
+		{"", false},
+		{"empty string", false},
+		{"_404_404", true},
+		{"FOO_BAR", true},
+		{"FOOBAR", false},
+		{"FO", false},
+		{"F_O", false},
+		{"_FOOBAR", true},
+		{"FOOBAR_", true},
+		{"FOO123", false},
+		{"FOO_123", true},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.varName, func(t *testing.T) {
+			if got := isUpperUnderscore(tt.varName); got != tt.expected {
+				t.Errorf("isUpperUnderScore(%s) = %v; want %v", tt.varName, got, tt.expected)
+			}
+		})
+	}
+}
+func TestIsDigitFunction(t *testing.T) {
+	tests := []struct {
+		input    rune
+		expected bool
+	}{
+		{'0', true},
+		{'1', true},
+		{'2', true},
+		{'3', true},
+		{'4', true},
+		{'5', true},
+		{'6', true},
+		{'7', true},
+		{'8', true},
+		{'9', true},
```

Comment:
Please add a unicode digit here

You can find one in the documentation of unicode.IsDigit